Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docrepository/adapter/dao/DocumentDao.java

```diff
@@ -102,176 +104,179 @@ protected Session getSession() {
+
+                String patientId = parameters.getPatientId();
                 if (patientId != null) {
-                    criteria.add(Expression.eq("patientId", patientId));
+                    criteria.add(Restrictions.eq("patientId", patientId));
                 }
 
-                if (classCodes != null && !classCodes.isEmpty()) {
-                    /**************************************************************
-                     * The class code and class code scheme combination can come in two different formats:
-                     *
-                     * <ns7:Slot name="$XDSDocumentEntryClassCode"> <ns7:ValueList> <ns7:Value>34133-9</ns7:Value>
-                     * </ns7:ValueList> </ns7:Slot> <ns7:Slot name="$XDSDocumentEntryClassCodeScheme"> <ns7:ValueList>
-                     * <ns7:Value>2.16.840.1.113883.6 .1</ns7:Value> </ns7:ValueList> </ns7:Slot>
-                     *
-                     * or
-                     *
-                     * <ns7:Slot name="$XDSDocumentEntryClassCode"> <ns7:ValueList> <ns7:Value>(
-                     * '34133-9^^2.16.840.1.113883.6.1')</ns7:Value> </ns7:ValueList> </ns7:Slot>
-                     *
-                     * The code below can deal with both formats.
-                     *
-                     *************************************************************/
-                    Criterion criterion = null;
-                    for (String classCode : classCodes) {
-                        if (LOG.isDebugEnabled()) {
-                            LOG.debug("Document query - class code: " + classCode);
-                        }
-                        String newClassCode;
-                        String newCodeScheme;
-
-                        if (classCode.contains("^^")) {
-                            int index = classCode.indexOf("^^");
-                            newClassCode = classCode.substring(0, index);
-                            newCodeScheme = classCode.substring(index + 2);
-                        } else {
-                            newClassCode = classCode;
-                            newCodeScheme = classCodeScheme;
-                        }
-
-                        Criterion andCrit = Expression.eq("classCode", newClassCode);
-                        if (newCodeScheme != null && !newCodeScheme.isEmpty()) {
-                            andCrit = Restrictions.and(andCrit, Expression.eq("classCodeScheme", newCodeScheme));
-                        }
-                        if (criterion == null) {
-                            criterion = andCrit;
-                        } else {
-                            criterion = Restrictions.or(criterion, andCrit);
-                        }
-                    }
-                    criteria.add(criterion);
+                List<String> statuses = parameters.getStatuses();
+                if (statuses != null && !statuses.isEmpty()) {
```

Comment:
Use CollectionUtil.isnotEmpty so it can check for null and is not empty

Reply:
Good Catch. Missed this.